bsb/plotting.py

```
@@ -643,11 +651,37 @@ def hdf5_plot_spike_raster(spike_recorders, input_region=None, show=True):
             xaxis=dict(title_text="Time (ms)"), yaxis=dict(title_text="Cell (ID)")
         )
     )
-    sort_by_size = lambda d: {k: v for k, v in sorted(d.items(), key=lambda i: len(i[1]))}
+    if sorted_labels is None:
+        sort_by_size = lambda d: {
+            k: v for k, v in sorted(d.items(), key=lambda i: len(i[1]))
+        }
+        sorted_labels = sort_by_size(x).keys()
     start_id = 0
-    for label, x, y in [(label, x[label], y[label]) for label in sort_by_size(x).keys()]:
-        y = [yi + start_id for yi in y]
-        start_id += ids[label]
+
+    # return x, y
+    for label, x, y in [(label, x[label], y[label]) for label in sorted_labels]:
+
+        if sorted_labels is None:
+            y = [yi + start_id for yi in y]
+            start_id += ids[label]
+        else:
+            if len(y) > 0:
+                # # Mapping with linear function
+                # y = list(np.array(y) - np.min(y) + start_id)
+                # y = np.min(y) + np.round((len(set(y))/(np.max(y)-np.min(y)))*(np.array(y)-np.min(y)))
+
+                if sorted_ids is None or (label not in sorted_ids.keys()):
+                    # Mapping with zip
+                    a = dict(zip(list(set(y)), range(start_id, start_id + len(set(y)))))
+                else:
+                    a = dict(
+                        zip(
+                            sorted_ids[label],
+                            range(start_id, start_id + len(sorted_ids[label])),
+                        )
+                    )
```

ids that are not in `sorted_ids` will be removed from the raster, is this intended?

---

yes otherwise we get blank spaces between raster plots  blocks. Nice catch!

---

So if there are 100 cells, unsorted, and you give 50 sorted cells that means that you will only show 50 cells. You do not want to show the remaining unsorted cells below the sorted ones? (If this is not a useful case we shouldn't bother adding it atm)

---

Mmmm I would say that if you want to sort the cells you plot, you sort all of them (I don't think one could need to show some sorted cells and some not).

---

But in case someone accidentally gives a shorter list, they are now silently hidden; source of annoying bugs ;p Maybe a warning?

---

Okok a warning is fine....